widgets/src/main/java/org/pentaho/gwt/widgets/client/listbox/CustomListBox.java

```diff
@@ -1140,6 +1124,11 @@ public void itemSelected( ListItem listItem, Event event ) {
     } else {
       setSelectedItem( listItem );
     }
+
+    if ( visible == 1 && popupShowing ) { // Drop-down mode
+      togglePopup();
+    }
```

Comment:
Please, do not add comments at the end of the line. Above or inside the `if` would be fine.